miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/ManifestApiCache.kt

```diff
@@ -0,0 +1,42 @@
+package com.rakuten.tech.mobile.miniapp.storage
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.rakuten.tech.mobile.miniapp.MiniAppManifest
+import kotlin.Exception
+
+/**
+ * A caching class to read and store the [MiniAppManifest] per MiniApp's id & version id using [SharedPreferences].
+ */
+@Suppress("TooGenericExceptionCaught")
+internal class ManifestApiCache(
```

Comment:
I'm not really sure if the `storage` folder is the right place for this 🤔  This is where information about the downloaded Mini App is stored, so I feel like it's going to be confusing seeing both `DownloadedManifestCache` and `ManifestApiCache` here.

Maybe you could move this class to the `api` folder and name it something like `ManifestResponseCache`?